contributors/design-proposals/federated-resource-quota.md

```diff
@@ -0,0 +1,53 @@
+# Federated ResourceQuota Requirements & Design
+@weijinxu, @IrfanUrRehman, @JianhuiZ
+reviewed by  @quinton-hoole, @deepak-vijSummary
+
+In a single Kubernetes cluster there is a concept of Resource Quota that manages the resources by namespace so that user usages can be limited to their fair share of the resources. We want to expand this concept to the federation level so that users can treat Federation Control Plane as a “Single Pane of Glass” for seamlessly managing the Resource Quota  information across all the underlying federated clusters.
+
+## User scenario
+
+As a federation user and an admin of clusters, I want to be able to define an overall resource quota for some specific or all resources for a given federated namespace. The overall resource quota limit should be applicable across clusters and preferably enforced at federation control plane level.
+
+For example, let us say that we’ve specified 100 pods in the resource quota spec in the federation namespace, with multiple underlying federated clusters. Let us assume that 50 pods have been deployed already across the federation in a previous deployment run. In a subsequent deployment run, if a user requests additional 50+ more pods to be deployed, the deployment will be rejected as a whole by the federation control plane before deployment process actually starts. If user requests less than 50 pods, the deployment should succeed as per the deployment preferences.
+
+## Implementation details
+### Create Local Resource Quota in underlying clusters
```

Comment:
Yes makes sense. Actually we did try to put up a lot more detail in the design earlier, but it did seem to become quite long and unreadable. (Somehow our problem that we could not flush out the details in a concise manner).
Actually our intention was to handle this design in two portions indeed, which to me is encompasses ur suggestion.
- quota enforcement at admission in fcp (handles object count as in ur point 1, except the embedded objects, viz pods in rs and rs in deployment).
- quota enforcement run time (as in your point 2 above).

@weijinxu I see that you have a small section at the end (Checking admit policy on Federation Control Plane (FCP)), but probably the rest of the doc is still confusing wrt to the two enforcement mechanisms. I suggest you make these 2 mechanisms in 2 sections explicit and modify this doc and examples accordingly.